mcode/mcode-coverage-checker
# STEAM-1010/deploy-fixes

Fix auto-deployment issures with our application

# Description
Issue: [STEAM-1010](https://jira.mitre.org/browse/STEAM-1010)

This PR replaces CommonJS require/module.exports syntax with ES Modules import/export syntax across all our modules. This fixes an Import Error problem that would occur in the build pipeline – for some reason, modules imported into the application that used CommonJS syntax would be invalid in the build context. In transitioning entirely to ESM syntax, however, our test pipeline started to not work. Earlier `npm run test` commands would run Jest directly, which evidently did not have support for ESM syntax (this needs to be transpiled into ES5). Looking further into CRA documentation, however, I realized that we could use the `react-scripts test` command, which would perform all necessary transpiling, so long as the tests were wherever react-scripts expects them to be. Hence the renaming/relocation of our testing directory. 

## Testing Recommendations
While I've since reverted the deploy script back to its original state, earlier commits to the branch had triggered deployments of the most recent version of the code to github-pages. To ensure that the build works, look at the logs for those earlier builds. Additionally, do the following: 
- Ensure that the [mcode.github.io/mcode-coverage-checker/](mcode.github.io/mcode-coverage-checker/) looks as expected 
- Ensure that `npm runstart` still works as expected
- Ensure that `npm run buil` works locally
- Ensure that `npm run test` works as expected. 

# Checklists

**Submitter:**

@ACCT1 :
